src/mscorlib/src/System/Threading/SynchronizationContext.cs

```diff
@@ -244,7 +241,7 @@ private static SynchronizationContext GetThreadLocalContext()
             SynchronizationContext context = null;
             
 #if FEATURE_APPX
-            if (context == null && Environment.IsWinRTSupported)
+            if (context == null && Environment.IsWinRTSupported && AppDomain.IsAppXModel())
```

Comment:
You can remove the `Environment.IsWinRTSupported` check. `AppDomain.IsAppXModel` is sufficient.